src/components/lib/cos_kernel_api.c

```diff
@@ -109,7 +110,7 @@ __mem_bump_alloc(struct cos_compinfo *__ci, int km)
 		/* are we dealing with a kernel memory allocation? */
 		syscall_op_t op = km ? CAPTBL_OP_MEM_RETYPE2KERN : CAPTBL_OP_MEM_RETYPE2USER;
 
-		if (call_cap_op(ci->pgtbl_cap, op, ret, 0, 0, 0)) return 0;
+		if (call_cap_op(ci->local_mi.pgtbl_cap, op, ret, 0, 0, 0)) return 0;
```

Comment:
Once we do RETYPE this, this component has access to that page, should that mean it has to be copied to its `SELF_PT`? I'm confused what remains in `SELF_PT` otherwise.